From 3d16531634ce5f0754df025b0a92e947115c0fd4 Mon Sep 17 00:00:00 2001
From: Joey <joey@lineageos.org>
Date: Thu, 5 Apr 2018 15:21:52 +0200
Subject: [PATCH 28/28] LatinIME: support for incognito mode

Don't learn suggestions when input specifies the
IME_FLAG_NO_PERSONALIZED_LEARNING flag

Change-Id: I224c36deda560657996cffa2922392a64e45b53f
Signed-off-by: Joey <joey@lineageos.org>
---
 java/src/com/android/inputmethod/latin/InputAttributes.java   | 4 ++++
 .../android/inputmethod/latin/settings/SettingsValues.java    | 3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/java/src/com/android/inputmethod/latin/InputAttributes.java b/java/src/com/android/inputmethod/latin/InputAttributes.java
index 37effeead..22a6f18d5 100644
--- a/java/src/com/android/inputmethod/latin/InputAttributes.java
+++ b/java/src/com/android/inputmethod/latin/InputAttributes.java
@@ -43,6 +43,7 @@ public final class InputAttributes {
     final public boolean mApplicationSpecifiedCompletionOn;
     final public boolean mShouldInsertSpacesAutomatically;
     final public boolean mShouldShowVoiceInputKey;
+    final public boolean mNoLearning;
     /**
      * Whether the floating gesture preview should be disabled. If true, this should override the
      * corresponding keyboard settings preference, always suppressing the floating preview text.
@@ -86,6 +87,7 @@ public final class InputAttributes {
             mShouldShowVoiceInputKey = false;
             mDisableGestureFloatingPreviewText = false;
             mIsGeneralTextInput = false;
+            mNoLearning = false;
             return;
         }
         // inputClass == InputType.TYPE_CLASS_TEXT
@@ -139,6 +141,8 @@ public final class InputAttributes {
                 && InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD != variation
                 && InputType.TYPE_TEXT_VARIATION_WEB_EMAIL_ADDRESS != variation
                 && InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD != variation;
+
+        mNoLearning = (editorInfo.imeOptions & EditorInfo.IME_FLAG_NO_PERSONALIZED_LEARNING) != 0;
     }
 
     public boolean isTypeNull() {
diff --git a/java/src/com/android/inputmethod/latin/settings/SettingsValues.java b/java/src/com/android/inputmethod/latin/settings/SettingsValues.java
index fa5ae00c4..a5bbb6f1f 100644
--- a/java/src/com/android/inputmethod/latin/settings/SettingsValues.java
+++ b/java/src/com/android/inputmethod/latin/settings/SettingsValues.java
@@ -184,7 +184,8 @@ public class SettingsValues {
                 && prefs.getBoolean(Settings.PREF_GESTURE_FLOATING_PREVIEW_TEXT, true);
         mAutoCorrectionEnabledPerUserSettings = mAutoCorrectEnabled
                 && !mInputAttributes.mInputTypeNoAutoCorrect;
-        mSuggestionsEnabledPerUserSettings = readSuggestionsEnabled(prefs, res);
+        mSuggestionsEnabledPerUserSettings = !mInputAttributes.mNoLearning &&
+                readSuggestionsEnabled(prefs, res);
         mIsInternal = Settings.isInternal(prefs);
         mHasCustomKeyPreviewAnimationParams = prefs.getBoolean(
                 DebugSettings.PREF_HAS_CUSTOM_KEY_PREVIEW_ANIMATION_PARAMS, false);
-- 
2.17.1

